src/Sulu/Bundle/TranslateBundle/Controller/CodeController.php

```diff
@@ -279,8 +279,24 @@ public function putAction(Request $request, $id)
             $code->setBackend($backend);
             $code->setFrontend($frontend);
             $code->setLength($length);
-            $code->setPackage($em->getReference(self::$packageEntity, $package['id']));
-            $code->setLocation($em->getReference(self::$locationEntity, $location['id']));
+
+            $package = $em->find(self::$packageEntity, $package['id']);
+            $location = $em->find(self::$locationEntity, $location['id']);
+
+            if (null === $package) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Package "%s" not found', $package['id']
+                ));
+            }
+
+            if (null === $location) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Location "%s" not found', $location['id']
+                ));
+            }
+
+            $code->setPackage($package);
+            $code->setLocation($location);
```

Comment:
Not sure if this should be kept. Although this was a Sqlite work-around, I do not think it is a *bad* thing to explicitly check for the existence of the package / location. The error response is clearer and this is not performance critical.